pkg/agent/agent.go

```diff
@@ -518,18 +525,32 @@ func (i *Initializer) setupDefaultTunnelInterface() error {
 			tunnelPortName = defaultTunInterfaceName
 			i.nodeConfig.DefaultTunName = tunnelPortName
 		}
-		tunnelPortUUID, err := i.ovsBridgeClient.CreateTunnelPortExt(tunnelPortName, i.networkConfig.TunnelType, config.DefaultTunOFPort, localIPStr, "", "", nil)
+		tunnelPortUUID, err := i.ovsBridgeClient.CreateTunnelPortExt(tunnelPortName, i.networkConfig.TunnelType, config.DefaultTunOFPort, i.networkConfig.TunnelCsum, localIPStr, "", "", nil)
 		if err != nil {
 			klog.Errorf("Failed to create tunnel port %s type %s on OVS bridge: %v", tunnelPortName, i.networkConfig.TunnelType, err)
 			return err
 		}
-		tunnelIface = interfacestore.NewTunnelInterface(tunnelPortName, i.networkConfig.TunnelType, localIP)
+		tunnelIface = interfacestore.NewTunnelInterface(tunnelPortName, i.networkConfig.TunnelType, localIP, i.networkConfig.TunnelCsum)
 		tunnelIface.OVSPortConfig = &interfacestore.OVSPortConfig{tunnelPortUUID, config.DefaultTunOFPort}
 		i.ifaceStore.AddInterface(tunnelIface)
 	}
 	return nil
 }
 
+func (i *Initializer) setTunnelCsum(tunnelPortName string, csum bool) error {
+	options, err := i.ovsBridgeClient.GetInterfaceOptions(tunnelPortName)
+	if err != nil {
+		return fmt.Errorf("error getting interface options: %w", err)
+	}
+
+	updatedOptions := make(map[string]interface{})
+	for k, v := range options {
+		updatedOptions[k] = v
+	}
+	updatedOptions["csum"] = strconv.FormatBool(csum)
+	return i.ovsBridgeClient.SetInterfaceOptions(tunnelPortName, updatedOptions)
```

Comment:
I guess that based on our current usage of this function, a race condition because of the separate Get / Set steps is not possible, so that's fine by me.